rstudio/rstudio
# Shuffle code from rstudio-pro to rstudio for adding users to the DB

### Intent

Part of [Pro #3094](https://github.com/rstudio/rstudio-pro/issues/3094): add users to the database in Open Source to support foreign key relationship.

NOTE: please do not merge, this requires a Pro-side companinion

### Approach

Shuffle code out of Pro and into OS

### Automated Tests

We should consider adding a test to check if the user entry is in the database after login, but it's not done yet.

### QA Notes

Once you've logged in with a user, attach to the database and check via `SELECT * FROM licensed_users;`. If you log out and log back in, you should also see the last sign-in time column get updated.

For open source, users should never be locked and never be admins. If the values in the database are updated externally, that should still have no impact on the behaviour (i.e. they're ignored).

### Checklist
